src/epanet.c

```diff
@@ -2803,19 +2818,39 @@ int DLLEXPORT EN_getnumdemands(EN_Project p, int nodeIndex, int *numDemands)
 **----------------------------------------------------------------
 */
 {
-    //Pdemand d;
-    //int n = 0;
+    // Check for valid arguments
+    *numDemands = 0;
+    if (!p->Openflag) return 102;
+    if (nodeIndex <= 0 || nodeIndex > p->network.Nnodes) return 203;
+    if (nodeIndex > p->network.Njuncs) return 0;
+
+    // Return number of demand categories
+	*numDemands = get_num_demands(p->network.Node[nodeIndex].D);
+    return 0;
+}
 
+int DLLEXPORT EN_getdemandindex(EN_Project p, int nodeIndex, const char *name,
```

Comment:
Unnecessary. Does not make sense to index a linked list.

Reply:
It is necessary, as keys won't work in the current context, and both require list traversal to find the data requested.